controllers/controllers_test.go

```diff
@@ -2,85 +2,117 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
 	"github.com/bombsimon/logrusr"
-	"github.com/go-logr/logr"
 	cassdcapi "github.com/k8ssandra/cass-operator/operator/pkg/apis/cassandra/v1beta1"
 	api "github.com/k8ssandra/k8ssandra-operator/api/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/clientcache"
 	"github.com/k8ssandra/k8ssandra-operator/test/framework"
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/rand"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
-	"path/filepath"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"testing"
-	"time"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
-var cfg *rest.Config
-var testClient client.Client
-var testEnv *envtest.Environment
-
 const (
-	timeout  = time.Second * 10
-	interval = time.Millisecond * 250
+	clustersToCreate = 2
+	timeout          = time.Second * 10
+	interval         = time.Millisecond * 250
+)
+
+var (
+	cfgs        = make([]*rest.Config, clustersToCreate)
+	testClients = make([]client.Client, clustersToCreate)
```

Comment:
`testClients` could be a map, it would be simpler because you could pass it directly to the `NewFramework` function:

```
f := framework.NewFramework(testClients["cluster-0"], "cluster-0", testClients)
```

```suggestion
	testClients = make(map[string]client.Client, clustersToCreate)
```